src/utils/utils.ts

```diff
@@ -152,6 +150,35 @@ export async function getGitBranchName() {
   return gitApi.repositories[0]?.state.HEAD?.name;//TODO: replace with getFromState(context, constants.branchName) when the onBranchChange is working properly
 }
 
+function extractRepoFullName(remoteURL: string): string | undefined {
+  const match = remoteURL.match(/[:/]([^/]+)\/([^/]+?)(?:\.git)?$/);
+  return match ? `${match[1]}/${match[2]}` : undefined;
+}
+
+export async function getActiveRepository(): Promise<Repository | undefined> {
+  const gitAPI = await getGitAPIRepository();
+  if (!gitAPI) {
+    return undefined;
+  }
+  return gitAPI.repositories[0]; // Default to the first repository
+}
+
+export async function getRepositoryFullName(): Promise<string | void> {
```

Comment:
change return type to Promise<string | undefined>, to ensures consistency with the return type of extractRepoFullName